cpp/dolfinx/io/XDMFFile.h

```diff
@@ -170,10 +170,11 @@ class XDMFFile
   /// GridType="Uniform"
   /// @param[in] attribute_name Name of the attribute to read
   /// @param[in] xpath XPath where MeshTags Grid is stored in file
-  mesh::MeshTags<std::int32_t>
-  read_meshtags(const mesh::Mesh<double>& mesh, std::string name,
-                std::optional<std::string> attribute_name,
-                std::string xpath = "/Xdmf/Domain");
+  template <typename T = std::int32_t>
```

Comment:
What types are supported? If the type in the file can't be checked it can lead to all sorts of crashes and mis-behaviours.

What is needed for this to be robust is a function for querying the data type in the file. I started this ages ago for HDF5 files, but it wasn't trivial to make it precise, but getting the type size was easy and may be adequate.

Reply:
We hit the same issue with the recent checkpointing project. We should have a chat in #developmemt about the best ways to tackle this.